### ISSUE TITLE
Breaks with eleventyComputed.js

### ISSUE THREAD

**Issue by DannyJoris** [kentaroi/eleventy-sass#13]
Bit of a bizarre bug, but eleventy-sass seems to break whenever I have a global `eleventyComputed.js` file present. It can be the simplest export like:
```
module.exports = {
  test: {
    foobar: 'test',
  },
};
```
Then eleventy-sass will break with the following message after I run:
```
DEBUG=*EleventySass* npx @11ty/eleventy --watch --serve --port=8080
```
```
  Dev:EleventySass getCacheKey(contents, ./src/assets/css/styles.scss) -> src/assets/css/styles.scss2022-08-09T04:14:53.746Zinitial +0ms
  Dev:EleventySass getCacheKey(contents, ./src/assets/css/styles.scss) -> src/assets/css/styles.scss2022-08-09T04:14:53.746Zinitial +26ms
  Dev:EleventySass getCacheKey(contents, ./src/assets/css/styles.scss) -> src/assets/css/styles.scss2022-08-09T04:14:53.746Zinitial +0ms
  Dev:EleventySass getCacheKey(contents, ./src/assets/css/styles.scss) -> src/assets/css/styles.scss2022-08-09T04:14:53.746Zinitial +6ms
[11ty] Problem writing Eleventy templates:
[11ty] 1. Having trouble rendering scss template ./src/assets/css/styles.scss (via TemplateContentRenderError)
[11ty] 2. Having trouble compiling template ./src/assets/css/styles.scss (via TemplateContentCompileError)
[11ty] 3. expected "{".
[11ty]   ╷
[11ty] 1 │ test
[11ty]   │     ^
[11ty]   ╵
[11ty]   src/assets/css/styles.scss 1:5  root stylesheet (via Error)
[11ty] Wrote 0 files in 1.29 seconds (v1.0.1)
```

If I remove eleventy-sass by removing `eleventyConfig.addPlugin(eleventySass);`, then everything compiles fine again.

Using global data from json files, works fine.

I have it reproduced here: https://github.com/DannyJoris/eleventy-sass-bug

**Comment by DannyJoris** [comment]
I can work around the issue easily by changing the file and format to: `_data/test.js`:
```
module.exports = {
  foobar: 'test',
};
```
Perhaps it doesn't make sense to use `eleventyComputed.js` globally anyways? Either way I'll leave this issue open just in case.

**Comment by kentaroi** [comment]
Thank you very much for filing the issue and providing the minimal project.

I briefly investigated it and am guessing that it is a bug ( or a spec) in Eleventy.

If you have `eleventyComputed.js` file, Eleventy will get computed data before rendering templates.

And here is the source of this bug:
https://github.com/11ty/eleventy/blob/0bbc16dfc03eb773de33488dd1c7b1bb37f701be/src/Template.js#L532-L563

If you have the following `eleventyComputed.js`:
```JavaScript
module.exports = {
  test: {
    foobar: () => 'test',
  },
};
```
Eleventy executes L535 and won't crash, because the value of `test.foobar` is a function.
https://github.com/11ty/eleventy/blob/0bbc16dfc03eb773de33488dd1c7b1bb37f701be/src/Template.js#L535

However, if you have the following `eleventyComputed.js`:
```JavaScript
module.exports = {
  test: {
    foobar: 'test',
  },
};
```

Eleventy executes L551-L558, because the value of `test.foobar` is of type String.
https://github.com/11ty/eleventy/blob/0bbc16dfc03eb773de33488dd1c7b1bb37f701be/src/Template.js#L551-L558

In L551-L588, Eleventy calls `this.renderComputedData()`, and `renderComputedData()` calls `_render()` with 'test', the value of `test.foobar`, as the first argument.
https://github.com/11ty/eleventy/blob/0bbc16dfc03eb773de33488dd1c7b1bb37f701be/src/TemplateContent.js#L356-L362

In short, `eleventy-sass` tried and failed to compile 'test', the value of `test.foobar`, as if it is a Sass/SCSS file content.

I'm not sure why Eleventy was implemented like that, but making properties in `eleventyComputed.js` have function values should work.

**Comment by kentaroi** [comment]
To be honest, Eleventy ought to fix the issue, but I fixed it in the plugin.
https://github.com/kentaroi/eleventy-sass/commit/93e3123e80092e95d18e5e6af30965956c480fd5

The fix was just published as v2.1.2.
